virttest/utils_net.py

```diff
@@ -3087,38 +3096,43 @@ def verify_ip_address_ownership(ip, macs, timeout=20.0, devs=None):
     :param devs: A set of network interfaces to check on. If is absent
                  then use route table to get a name for possible
                  network interfaces.
+    :param session: ShellSession object of remote host
     :return: True if ip is assigned to a MAC address in macs.
     """
-    def __arping(ip, macs, dev, timeout):
+    def __arping(ip, macs, dev, timeout, session=None):
+        func = process.getoutput
+        if session:
+            func = session.cmd_output
         # Compile a regex that matches the given IP address and any of the
         # given MAC addresses from arping output
-        ip_map = parse_arp()
+        ip_map = parse_arp(session=session)
         for mac in macs:
             if ip_map.get(mac) == ip:
                 return True
 
         mac_regex = "|".join("(%s)" % mac for mac in macs)
         regex = re.compile(r"\b%s\b.*\b(%s)\b" % (ip, mac_regex), re.I)
         arping_bin = utils_path.find_command("arping")
+        if session:
+            arping_bin = func("which arping")
         cmd = "%s --help" % arping_bin
-        if "-C count" in decode_to_text(process.system_output(cmd, shell=True, ignore_status=True,
-                                                              verbose=False)):
+        if "-C count" in func(cmd):
             regex = re.compile(r"\b%s\b.*\b(%s)" % (mac_regex, ip), re.I)
             arping_cmd = "%s -C1 -c3 -w%d -I %s %s" % (arping_bin, int(timeout),
                                                        dev, ip)
         else:
             arping_cmd = "%s -f -c3 -w%d -I %s %s" % (arping_bin, int(timeout),
                                                       dev, ip)
         try:
-            o = decode_to_text(process.system_output(arping_cmd, shell=True, verbose=False))
-        except process.CmdError:
+            o = func(arping_cmd)
+        except (process.CmdError, aexpect.ShellError, aexpect.ShellTimeoutError):
```

Comment:
Here `aexpect.ShellError` is enough, it should cover `aexpect.ShellTimeoutError` and all other sub-type.

Reply:
cool, I will make the suggested change.